.github/workflows/build-test-actions.yml

```diff
@@ -24,8 +24,10 @@ jobs:
       PYTHON: ${{ matrix.python-version }}
     steps:
       - uses: actions/checkout@v4
-      - uses: chartboost/ruff-action@v1
-      # https://github.com/chartboost/ruff-action
+      - uses: astral-sh/ruff-action@v3
+        with:
+          args: "format --check --diff"
+      # https://github.com/astral-sh/ruff-action
       - name: Set up Python ${{ matrix.python-version }}
```

Comment:
These changes fix https://github.com/vprusso/toqito/pull/1161#issuecomment-2787944941 

if the formatter catches a file that needs to be reformatted, this workflow will fail.